src/main/java/com/unb/tracker/validator/CourseValidator.java

```diff
@@ -0,0 +1,33 @@
+package com.unb.tracker.validator;
+
+import com.unb.tracker.model.Course;
+import com.unb.tracker.service.CourseService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.validation.Errors;
+import org.springframework.validation.ValidationUtils;
+import org.springframework.validation.Validator;
+
+@Component
+public class CourseValidator implements Validator {
+    @Autowired
+    private CourseService courseService;
+
+    @Override
+    public boolean supports(Class<?> aClass) {
+        return Course.class.equals(aClass);
+    }
+
+    @Override
+    public void validate(Object o, Errors errors) {
+        Course course = (Course) o;
+
+        if (course.getRows() < 1 || course.getRows() > 18) {
```

Comment:
Should these values be pulled out into a config file? I'm not sure how it would be done but it might save some headaches later

Reply:
I +1 this idea but believe it should be looked after in some sort of improvement issue.

Reply:
I agree with mike. I was going to do a similar thing in the auth issue, but didn't get around to it. We could create an issue to do this for all types of validation--user and course.